### ISSUE TITLE
Improving the confirmation rate of alert subscriptions

### ISSUE THREAD

**Issue by evansd** [ebmdatalab/openprescribing#1048]
Currently we have 2037 organisation bookmarks and of these 1262 are approved (i.e. the recipient clicked the link saying "Yes I want these"). This rate of 62% can probably be improved.

1. Subscription UX is a bit weird. If I go to a CCG page, enter my email address, tick both the Newsletter and Monthly Alerts boxes and hit Subscribe I am taken to a form which asks for some extra details for the newsletter. If I complete this form I get two green confirmation notices, one saying that I'm signed up to the newsletter and the other saying I am subscribed to monthly alerts. However, I'm not subscribed to monthly alerts until I click the link in the confirmation email.

2. Can we do things with mailgun to improve our spam score? E.g. are we using shared IPs, and can we switch to dedicated so we can build up our own trust reputation?

3. A thing I've implemented before is to, when a user subscribes, store something in the session with their email address and the fact that they have a pending confirmation. On each page load you can check whether this email address is confirmed yet and, if not, display a banner at the top of the page saying "You signed up for alerts with foo@bar.com, please check your email and click the confirmation link". It's a bit more work but it stops people thinking they'e subscribed when they haven't.

4. A few more suggestions based on initial discussion with @HelenCEBM:
* Can we warn people when they subscribe that they'll need to check their junk/spam folders.
* Can we do a one-off mail to all the people who signed up but didn't confirm in the last X months?
* Do we need to require confirmation at all? Is an opt-out sufficient? *(Dave: I think we probably do need the opt in on GDPR grounds)*

**Comment by HelenCEBM** [comment]
Easiest first step could be to change the subject of the confirmation emails to say "Please confirm your subscription". 

I don't think GDPR prevents us doing one-step rather than 2-step opt-in, so I think we should explore this option. 
- I can check our mailchimp settings for the newsletter.
- Do we need some control in place (if possible) if the same email is entered a large number of times, to protect users against some joker putting someone else's email in??

**Comment by HelenCEBM** [comment]
Update: Newsletter now has a welcome email but only a one-step signup

**Comment by sebbacon** [comment]
Partial dupe of #721

**Comment by sebbacon** [comment]
Possible quick wins:

* We could definitely fix the weird banner situation.
* We're already setting a cookie somewhere in the signup process, IIRC, so the suggestion "check your email" is not only sensible, but shouldn't be *too* tough.
* I see no reason not to do a one-off mail. I've actually done that before, about 18 months ago now I think.

Possible harder wins thoughts:

## Dedicated IP

I'm not sure of the potential returns in using dedicated IP - the change is not risk-free, and monitoring it comes with an extra cost.  The June mailout only had a 1.9% rejection rate, so it's probably the final thing we should worry about:

```
┌────────────┬───────┐
│ event_type │ count │
├────────────┼───────┤
│ bounced    │     2 │
│ clicked    │   440 │
│ delivered  │  1303 │
│ opened     │  1790 │
│ rejected   │    67 │
└────────────┴───────┘
```


Mailgun's [advice on this](https://documentation.mailgun.com/en/latest/best_practices.html#ip-addresses-and-sending-volume) seems sensible:

> .. if you are a high volume sender you should consider getting a pool of IPs. However, your reputation can also be hurt if you are not sending enough volume consistently from an IP so it’s a tricky balance.
> 
> If your email sending is volatile with large spikes of volume, ESPs may assume those large spikes are spam. Also, if you overall volume is too low, they won’t acknowledge your reputation. Generally, if you are sending less than 5,000 emails per day, a shared IP may be the right solution.


## Double opt-in

Reasons for double opt-in:

 * prevent people from signing up other people (either well-meaning or black- at, i.e. signing up 100000 alerts is a convenient route to a monthly DoS of our website)
 * check the address is right, so we don't send email unnecessarily

As there's an unsubscribe link on every email, I suppose the well-meaning 3rd party signup scenario isn't really something to worry about.

So it's really just the doing-unnecessary-work-possible-DoS scenario we might need to worry about.

We're already recording bounces in our `Maillog` model, so it would be trivial to deal with bounces, just leaving us with the DoS scenario.

Actually, at the moment, I'm concerned we're not tracking mail sends very well in any case; key leakage and/or bad staging config has in the past led to unexpected £300+ monthly bills.  

## Summary 

To summarise - I'm happy to drop double opt-in if we:

* Automatically unsubscribe/delete/similar bouncing addresses
* Instigate an alert when our rate of email subscription increases suddenly

That means we don't need to bother with improving the signup messages; we could remove them instead.

As a thought - for the alerts, this might ultimately plug into the dashboard we've been considering for the team. So we might consider using statsd and [New Relic's integration for that](https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/statsd-monitoring-integration)

**Comment by evansd** [comment]
All sounds very sensible to me. Good point about the IP issue.

Could we retroactively apply our single opt-in principle and email everyone who has signed up to alerts but not confirmed saying something like:

> In the past you signed up for monthly alerts from OpenPrescribing (a website which blah blah...) Due to a technical issue you weren't receiving these but this has now been corrected. You can unsubscribe at any time by clicking the relevant link in the alert emails.

**Comment by HelenCEBM** [comment]
User feedback: "I did subscribe ... but I have just found the confirmation email in my junk so have now confirmed it but this probably gives you a heads up that nhs.net is likely to divert to junk so might be worth giving people a notification that they need to check and follow the verification link."

**Comment by evansd** [comment]
@seb We've done the main things we discussed here. There's still weirdness around the signup UX but I think we decided that we don't have resources to tackle this at the moment so I suggest we just close this ticket. Agree?

**Comment by sebbacon** [comment]
Agree! Let's rip the whole thing out another time!
